src/simulator_core/infrastructure/plotting.py

```diff
@@ -0,0 +1,112 @@
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""Plot result on the map for debugging."""
+import esdl
+import plotly.express as px
+import geopandas as gpd
+from pandas import DataFrame
+import numpy as np
+from shapely.geometry import LineString
+from numpy.typing import NDArray
+from simulator_core.entities.esdl_object import EsdlObject
+
+
+class Plotting:
+    """Plotting plot the results of asset (currently pipe) for inspection."""
+
+    def __init__(self, esdl_object: EsdlObject) -> None:
+        """Initialize plotting instance.
+
+        :param EsdlObject esdl_object: esdl object of the network
+        """
+        self.esdl_object = esdl_object
+
+    def plot_map(self, result: DataFrame) -> None:
+        """Static function to plot the simulation results to the map.
+
+        :param DataFrame result: simulation result in dataframe format
+        """
+        geo_df = self.prepare_data(result)
+        Plotting.plot_line_mapbox(geo_df)
+
+    def prepare_data(self, result: DataFrame) -> gpd.GeoDataFrame:
+        """Function to prepare geopandas dataframe."""
+        geo_df = gpd.GeoDataFrame(
+            columns=['asset_id', 'asset_name', 'geometry', 'mass_flow', 'pressure_in',
+                     'pressure_out', 'temperature_in', 'temperature_out'], crs="EPSG:4326")
+
+        assets = self.get_assets_by_asset_type(esdl.Pipe)
+
+        for asset in assets:
+            if isinstance(asset.geometry, esdl.esdl.Line):
+                coor_pipe = []
+                for point in asset.geometry.point:
+                    coor_pipe.append([point.lon, point.lat])
+
+                last_data_index = len(result) - 1
+                data = {'asset_id': asset.id,
+                        'asset_name': asset.name,
+                        'geometry': LineString(coor_pipe),
+                        'mass_flow': result[(asset.id, 'mass_flow')][last_data_index],
+                        'pressure_in': result[(asset.id, 'pressure_supply')][last_data_index],
+                        'pressure_out': result[(asset.id, 'pressure_return')][last_data_index],
+                        'temperature_in': result[(asset.id, 'temperature_supply')][last_data_index],
+                        'temperature_out': result[(asset.id, 'temperature_return')][
+                            last_data_index],
+                        }
+
+                geo_df.loc[len(geo_df)] = data
+
+        return geo_df
+
+    @staticmethod
+    def plot_line_mapbox(geo_df: gpd.GeoDataFrame) -> None:
+        """Function to plot using plotly express line mapbox."""
+        lats: NDArray = np.array(None)
+        lons: NDArray = np.array(None)
+        names: NDArray = np.array(None)
+        for _, row in geo_df.iterrows():
+            if isinstance(row['geometry'], LineString):
```

Comment:
For the current implementation this check is unecessary right? Are not all pipes linestring elements?

Reply:
this is needed, because i only want asset that has linestring geometry